experimental/Schemes/BlowupMorphism.jl

```diff
@@ -0,0 +1,255 @@
+  CX = codomain(p_cov)
+  E = exceptional_divisor(p)
+  for U in patches(CY_simp)
+    p_res = p_cov_simp[U]
+    V = codomain(p_res)
+    J = image_ideal(inc)(V)
+    g = maps_with_given_codomain(inc, V)
+    pbJ = ideal(OO(U), pullback(p_res).(gens(J)))
+    pbJ_sat = saturated_ideal(pbJ)
+    pbJ_sat = saturation(pbJ_sat, ideal(base_ring(pbJ_sat), lifted_numerator.(E(U))))
+    pbJ = ideal(OO(U), [g for g in OO(U).(gens(pbJ_sat)) if !iszero(g)])
+    ID[U] = pbJ
+  end
+
+  I_trans = IdealSheaf(Y, ID, check=false) # TODO: Set to false
+  inc_Z_trans = CoveredClosedEmbedding(Y, I_trans, covering=CY_simp, check=false)
+  inc_cov = covering_morphism(inc_Z_trans)
+
+  Z_trans = domain(inc_Z_trans)
+  pr_res = restrict(projection(p), inc_Z_trans, inc)
+  return Z_trans, inc_Z_trans, pr_res
+end
+
+#=
+#  Z' ↪ Y
+#       ↓ f
+#  Z ↪  X
+#
+#  Assuming f(Z') ⊂ Z, we compute and return the restriction f : Z' → Z.
+=#
+function restrict(f::AbsCoveredSchemeMorphism,
+    inc_dom::CoveredClosedEmbedding,
+    inc_cod::CoveredClosedEmbedding;
+    check::Bool=true
+  )
+  f_cov = covering_morphism(f)
+  inc_dom_cov = covering_morphism(inc_dom)
+  inc_cod_cov = covering_morphism(inc_cod)
+
+  # We need to do the following.
+  # - Pass to a common refinement ref_cod in X that both 
+  #   f and inc_cod can restrict to.
+  # - Pass to a common refinement in Y
+  ref_cod, a, b = _register!(common_refinement(codomain(f_cov), codomain(inc_cod_cov)), codomain(f))
+  inc_cod_ref = restrict(inc_cod, ref_cod)
+  f_res = restrict(f, ref_cod)
+  ref_dom, aa, bb = _register!(common_refinement(domain(f_res), codomain(inc_dom_cov)), domain(f))
+  inc_dom_ref = restrict(inc_dom, ref_dom)
+  inc_dom_ref = compose(inc_dom_ref, aa)
+  # Collecting the maps for the restricted projection here
+  map_dict = IdDict{AbsSpec, AbsSpecMor}()
+  for U in patches(domain(inc_dom_ref))
+    q_res = compose(inc_dom_ref[U], f_res[codomain(inc_dom_ref[U])])
+    V = codomain(q_res)
+    g = maps_with_given_codomain(inc_cod_ref, V)
+    if !isone(length(g))
+      error()
+    end
+    pre_V = domain(first(g))
+    map_dict[U] = restrict(q_res, domain(q_res), pre_V, check=false)
+  end
+  psi = CoveringMorphism(domain(inc_dom_ref), domain(inc_cod_ref), map_dict, check=false)
+  return CoveredSchemeMorphism(domain(inc_dom), domain(inc_cod), psi)
+end
+
+function _register!(data::Tuple{<:Covering, <:CoveringMorphism, <:CoveringMorphism},
+    X::AbsCoveredScheme
+  )
+  push!(coverings(X), data[1])
+  refinements(X)[(domain(data[2]), codomain(data[2]))] = data[2]
+  refinements(X)[(domain(data[3]), codomain(data[3]))] = data[3]
+  return data
+end
+
+function maps_with_given_codomain(phi::CoveringMorphism, V::AbsSpec)
+  result = Vector{AbsSpecMor}()
+  for U in keys(morphisms(phi))
+    floc = morphisms(phi)[U]
+    codomain(floc) === V || continue
+    push!(result, floc)
+  end
+  return result
+end
+
+#function saturation(I::MPolyLocalizedIdeal, J::MPolyLocalizedIdeal)
```

Comment:
three saturation functions commented out: have they moved somewhere else and can bei removed?